### PR TITLE
Core/Packet Update Auction house, black market and banker opcode

### PR DESCRIPTION
**Pull request by ProjectcoreDevs**

### CODE REVIEW

File: src/server/game/Handlers/BlackMarketHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        GetPlayer()->RemoveAurasByType(SPELL_AURA_FEIGN_DEATH);
+
+    SendBlackMarketHello(packet.Guid, unit);
+}
+
+void WorldSession::SendBlackMarketHello(ObjectGuid Guid, Creature* unit)
+{
+    WorldPackets::BlackMarket::PlayerCliBlackMarketHello packet;
+    packet.Guid = Guid;
+    packet.Enable = true;
```
Comment (Shauren) [inline src/server/game/Handlers/BlackMarketHandler.cpp]: Questionable choice of value for an unimplemented feature...

File: src/server/game/Server/Packets/AuctionHousePackets.cpp [hunk 2]
```diff
@@ -19,6 +19,19 @@
...
 
+void WorldPackets::AuctionHouse::ClientAuctionHello::Read()
+{
+    _worldPacket >> Guid;
+}
+
+WorldPacket const* WorldPackets::AuctionHouse::PlayerCliAuctionHelloResponse::Write()
+{
+    _worldPacket << Guid;
+    _worldPacket.WriteBit(OpenForBusiness);
```
Comment (DDuarte) [inline src/server/game/Server/Packets/AuctionHousePackets.cpp]: Missing call to FlushBits() after WriteBit()

File: src/server/game/Server/Packets/BlackMarketPackets.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+void WorldPackets::BlackMarket::ClientBlackMarketHello::Read()
+{
+    _worldPacket >> Guid;
+}
+
+WorldPacket const* WorldPackets::BlackMarket::PlayerCliBlackMarketHello::Write()
+{
+    _worldPacket << Guid;
+    _worldPacket.WriteBit(Enable);
```
Comment (DDuarte) [inline src/server/game/Server/Packets/BlackMarketPackets.cpp]: Missing call to FlushBits() after WriteBit()

File: src/server/game/Server/Packets/BlackMarketPackets.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            PlayerCliBlackMarketHello() : ServerPacket(SMSG_BLACK_MARKET_HELLO, 15) { }
+
+            WorldPacket const* Write() override;
+
+            ObjectGuid Guid;
+            bool Enable = true;
+        };
+    }
+}
+#endif // BlackMarketPackets_h__
```
Comment (DDuarte) [inline src/server/game/Server/Packets/BlackMarketPackets.h]: Add an empty line to the end of this file

File: src/server/game/Server/Packets/NPCPackets.cpp [hunk 5]
```diff
@@ -118,3 +118,10 @@ WorldPacket const* WorldPackets::NPC::TrainerList::Write()
 
     return &_worldPacket;
 }
+
+WorldPacket const* WorldPackets::NPC::PlayerCliShowBank::Write()
+{
+    _worldPacket << Guid;
+
+    return &_worldPacket;
+}
```
Comment (DDuarte) [inline src/server/game/Server/Packets/NPCPackets.cpp]: Add an empty line to the end of this file

**Comment by Shauren** [comment]
Note: the agreed naming convention for packet classes is client class name with stripped prefixes like "PlayerCli", "Client" etc.

File: src/server/game/Server/Packets/AuctionHousePackets.cpp (hunk 2 above)
Comment (Shauren) [inline src/server/game/Server/Packets/AuctionHousePackets.cpp]: ALWAYS call FlushBits if a bit is last field in the packet

File: src/server/game/Server/WorldSession.h [hunk 6]
```diff
@@ -550,6 +560,9 @@ class WorldSession
         void SendAuctionOwnerNotification(AuctionEntry* auction);
         void SendAuctionRemovedNotification(uint32 auctionId, uint32 itemEntry, int32 randomPropertyId);
 
+        // Black Market
+        void SendBlackMarketHello(ObjectGuid Guid, Creature* unit);
```
Comment (DDuarte) [inline src/server/game/Server/WorldSession.h]: change `Guid` to `guid`
the 2nd argument is a unit or a creature? if this is a pointer to the auctioneer rename `unit` to `auctioneer`

File: src/server/game/Entities/Unit/Unit.h [hunk 7]
```diff
@@ -755,7 +755,8 @@ enum NPCFlags
     UNIT_NPC_FLAG_MAILBOX               = 0x04000000,       // mailbox
     UNIT_NPC_FLAG_REFORGER              = 0x08000000,       // reforging
     UNIT_NPC_FLAG_TRANSMOGRIFIER        = 0x10000000,       // transmogrification
-    UNIT_NPC_FLAG_VAULTKEEPER           = 0x20000000        // void storage
+    UNIT_NPC_FLAG_VAULTKEEPER           = 0x20000000,       // void storage
+    UNIT_NPC_FLAG_BLACK_MARKET          = 0x80000000        // black market
```
Comment (DDuarte) [inline src/server/game/Entities/Unit/Unit.h]: Unrelated to the pull request: any idea what is 0x40000000?

File: src/server/game/Server/Packets/AuctionHousePackets.cpp (hunk 2 above)
Comment (ProjectcoreDevs) [inline src/server/game/Server/Packets/AuctionHousePackets.cpp]: thanks :)

File: src/server/game/Server/Packets/BlackMarketPackets.h [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        {
+        public:
+            ClientBlackMarketHello(WorldPacket&& packet) : ClientPacket(CMSG_BLACK_MARKET_HELLO, std::move(packet)) { }
+
+            void Read() override;
+
+            ObjectGuid Guid;
+        };
+
+        class PlayerCliBlackMarketHello final : public ServerPacket
```
Comment (DDuarte) [inline src/server/game/Server/Packets/BlackMarketPackets.h]: Rename class to `BlackMarketHello`

File: src/server/game/Server/Packets/NPCPackets.h [hunk 9]
```diff
@@ -127,6 +127,16 @@ namespace WorldPackets
             int32 TrainerID     = 1;
             std::vector<TrainerListSpell> Spells;
         };
+
+        class PlayerCliShowBank final : public ServerPacket
```
Comment (DDuarte) [inline src/server/game/Server/Packets/NPCPackets.h]: Rename class to ShowBank

File: src/server/game/Server/Packets/BlackMarketPackets.h [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "Packet.h"
+#include "ObjectGuid.h"
+#include "WorldSession.h"
+
+namespace WorldPackets
+{
+    namespace BlackMarket
+    {
+        class ClientBlackMarketHello final : public ClientPacket
```
Comment (DDuarte) [inline src/server/game/Server/Packets/BlackMarketPackets.h]: Rename class to `BlackMarketHelloClient`

File: src/server/game/Server/Packets/AuctionHousePackets.h [hunk 11]
```diff
@@ -26,6 +26,26 @@ namespace WorldPackets
 {
     namespace AuctionHouse
     {
+        class ClientAuctionHello final : public ClientPacket
+        {
+        public:
+            ClientAuctionHello(WorldPacket&& packet) : ClientPacket(CMSG_AUCTION_HELLO, std::move(packet)) { }
```
Comment (DDuarte) [inline src/server/game/Server/Packets/AuctionHousePackets.h]: Rename class to `AuctionHello`

File: src/server/game/Server/Packets/AuctionHousePackets.h [hunk 12]
```diff
@@ -26,6 +26,26 @@ namespace WorldPackets
...
+            
+            void Read() override;
+            ObjectGuid Guid;
+        };
+
+        class PlayerCliAuctionHelloResponse final : public ServerPacket
+        {
+        public:
+            PlayerCliAuctionHelloResponse() : ServerPacket(SMSG_AUCTION_HELLO, 1 + 16) { }
```
Comment (DDuarte) [inline src/server/game/Server/Packets/AuctionHousePackets.h]: Rename class to `AuctionHelloResponse`

File: src/server/game/Entities/Unit/Unit.h (hunk 7 above)
Comment (cemak) [inline src/server/game/Entities/Unit/Unit.h]: UNIT_NPC_FLAG_BATTLE_PET            = 0x40000000, // battle pet

File: src/server/game/Server/Packets/AuctionHousePackets.h (hunk 11 above)
Comment (Vincent-Michael) [inline src/server/game/Server/Packets/AuctionHousePackets.h]: nope rename in BlackMarketOpen

File: src/server/game/Server/Packets/AuctionHousePackets.h (hunk 12 above)
Comment (Vincent-Michael) [inline src/server/game/Server/Packets/AuctionHousePackets.h]: nope rename in BlackMarketOpenResult

Comment (ProjectcoreDevs) [inline src/server/game/Server/Packets/AuctionHousePackets.h]: it's auction house this file :)

File: src/tools/map_extractor/loadlib/DBFilesClientList.h [hunk 13]
```diff
@@ -425,6 +425,12 @@ char const* DBFilesClientList[] =
     "DBFilesClient\\gtBattlePetTypeDamageMod.dbc",
     "DBFilesClient\\gtBarberShopCostBase.dbc",
     "DBFilesClient\\gtArmorMitigationByLvl.dbc",
+    "DBFilesClient\\gtNpcTotalHp.dbc",
```
Comment (DDuarte) [inline src/tools/map_extractor/loadlib/DBFilesClientList.h]: These changes should not be in this PR. See warning at the bottom of http://collab.kpsn.org/pages/viewpage.action?pageId=12714394

Comment (ProjectcoreDevs) [inline src/tools/map_extractor/loadlib/DBFilesClientList.h]: ok...i've revert it

**Pull request closed without merging**